src/AstComponentChecker.cpp

```diff
@@ -0,0 +1,223 @@
+/*
+ * Souffle - A Datalog Compiler
+ * Copyright (c) 2018, Oracle and/or its affiliates. All rights reserved
```

Comment:
If this is new code, then it was written by The Souffle Developers. If it's old code, moved, then the date is the same as the previous code.